### PR TITLE
[USM] add Redis decoding skeleton

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

This PR adds the "skeleton" for implementing Redis decoding in the future. This includes adding:
- Configuration options to enable Redis decoding
- a new protocol in USM, with its own statkeeper, map cleaner, etc..
- Dummy eBPF probes that will be dispatched to if classifying a Redis connection and decoding is enabled

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/redis/decoding.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+SEC("socket/redis_process")
+int socket__redis_process(struct __sk_buff *skb) {
+    return 0;
+}
+
+SEC("uprobe/redis_tls_process")
+int uprobe__redis_tls_process(struct pt_regs *ctx) {
+    return 0;
+}
```
Comment (guyarb) [inline pkg/network/ebpf/c/protocols/redis/decoding.h:14]: what about a stub for termination?

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

File: pkg/network/protocols/redis/protocol.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	p.mapCleaner.Stop()
+
+	if p.eventsConsumer != nil {
+		p.eventsConsumer.Stop()
+	}
+}
+
+// DumpMaps dumps map contents for debugging.
+func (p *protocol) DumpMaps(w io.Writer, mapName string, currentMap *ebpf.Map) {
+	if mapName == inFlightMap { // maps/postgres_in_flight (BPF_MAP_TYPE_HASH), key ConnTuple, value EbpfTx
```
Comment (guyarb) [inline pkg/network/protocols/redis/protocol.go:130]: maps/postgres

File: pkg/network/protocols/redis/protocol.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		Stats: nil,
+	}
+}
+
+// IsBuildModeSupported returns always true, as Redis module is supported by all modes.
+func (*protocol) IsBuildModeSupported(buildmode.Type) bool {
+	return true
+}
+
+func (p *protocol) processRedis(_ []EbpfEvent) {
```
Comment (guyarb) [inline pkg/network/protocols/redis/protocol.go]: processRedis(_ []EbpfEvent) > processRedis([]EbpfEvent)

File: pkg/network/protocols/redis/statskeeper.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+package redis
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+)
+
+type StatsKeeper struct {
```
Comment (guyarb) [inline pkg/network/protocols/redis/statskeeper.go:16]: documentation

File: pkg/network/protocols/redis/statskeeper.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+func NewStatsKeeper(c *config.Config) *StatsKeeper {
+	statsKeeper := &StatsKeeper{
+		maxEntries: c.MaxRedisStatsBuffered,
+	}
+
+	statsKeeper.resetNoLock()
+	return statsKeeper
+}
+
+func (s *StatsKeeper) Process(_ *EbpfEvent) {
```
Comment (guyarb) [inline pkg/network/protocols/redis/statskeeper.go:32]: documentation

File: pkg/network/protocols/redis/statskeeper.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	s.statsMutex.Lock()
+	defer s.statsMutex.Unlock()
```
Comment (guyarb) [inline pkg/network/protocols/redis/statskeeper.go:34]: redundant while the implementation is empty

**Review by guyarb — CHANGES_REQUESTED** [review]
Compared to https://github.com/DataDog/datadog-agent/pull/25416

there are still misses
please go over commit-commit to see the differences

File: pkg/network/ebpf/c/protocols/redis/types.h [hunk 7]
```diff
@@ -0,0 +1,19 @@
+#ifndef __REDIS_TYPES_H
+#define __REDIS_TYPES_H
+
+#include "conn_tuple.h"
+
+// Controls the number of Redis transactions read from userspace at a time.
+#define REDIS_BATCH_SIZE 25
```
Comment (guyarb) [inline pkg/network/ebpf/c/protocols/redis/types.h:7]: I'd start with the highest value possible, so we can always ensure we're maximizing the batches functionality

File: go.mod [hunk 8]
```diff
@@ -595,7 +595,7 @@ require github.com/lorenzosaino/go-sysctl v0.3.1
 
 require (
 	github.com/DATA-DOG/go-sqlmock v1.5.2
-	github.com/DataDog/agent-payload/v5 v5.0.124
+	github.com/DataDog/agent-payload/v5 v5.0.125-0.20240715134148-a4cf7e9f7d52
```
Comment (guyarb) [inline go.mod:598]: needs to be merged

**Review by guyarb — APPROVED** [review]

**Review by gjulianm — APPROVED** [review]
LGTM for ebpf-platform files

**Review by derekwbrown — APPROVED** [review]
approved for (2) files owned by wkit

**Review by hmahmood — APPROVED** [review]

**Pull request merged**